### PR TITLE
No marker no zoom

### PR DESCRIPTION
**Pull request by SiggyF**
- Sync zoom with navigation (zoom from current camera position to location of activeMarker)
- Fix coordinate problems (use consistent coordinate conversion)
- Rotate camera instead of scene (z is up, for now positive down, don't know why....)

### CODE REVIEW

File: components/globe-component/globe-component.js [hunk 1]
```diff
@@ -160,29 +170,10 @@ export default {
...
-            const cart = polar2cartesian(radius, latitude, longitude)
-            this.camera.position.set(cart.x, cart.y, cart.z)
-
-            this.camera.lookAt(new THREE.Vector3(0, 0, 0))
-          })
-          .easing(Easing.Cubic.InOut)
-          .start()
+        const { data } = this.intersections[0].object
+        // navigate to path
+        console.log('navigating to', data)
```
Comment (jurgenbelien) [inline components/globe-component/globe-component.js]: I think I put it here, but let's remove it ;-)

File: components/globe-component/globe-component.js [hunk 2]
```diff
@@ -281,7 +296,19 @@ export default {
       const renderWidth = this.containerSize[0]
       const renderHeight = this.containerSize[1] * (1 + vOffsetFactor)
       const camera = new THREE.PerspectiveCamera(30, renderWidth / renderHeight, 0.1, 300)
-      camera.position.z = 30 * (1 + vOffsetFactor) * (1 + vOffsetFactor)
+
+      // lat, lon in radians
+      const θ = lat2rad(40)
+      const φ = lon2rad(40)
```
Comment (jurgenbelien) [inline components/globe-component/globe-component.js]: also, change to `el`, `azi`?

File: components/globe-component/particles.js [hunk 3]
```diff
@@ -61,8 +61,10 @@ class Particles {
   }
 
   load (finished) {
-    const that = this
-
+    // Load data in format:
+    // -89.772727273,-60.0,0.0,0.0,0.0,0.0,0.0,0.0
+    // -89.772727273,60.0,0.0,0.0,0.0,0.0,0.0,0.0
+    // -89.318181818,-140.0,0.0,0.0,0.0,0.0,0.0,0.0
     loadData('https://s3-eu-west-1.amazonaws.com/deltares-opendata/wwa/wri/land.csv', (err, result) => {
```
Comment (jurgenbelien) [inline components/globe-component/particles.js]: this file is a whopping 4,5MB / 648KB Gzipped  😱 
Out of scope for this PR, but we should take a look at it nonetheless

File: components/globe-component/globe-component.js [hunk 4]
```diff
@@ -191,17 +182,43 @@ export default {
...
+    panToActiveMarker () {
+      if (this.activeMarker && this.activeMarker.location) {
+        const from = cartesian2polar(this.camera.position.x, this.camera.position.y, this.camera.position.z)
+        from.lat = θ2deg(from.θ)
+        from.lng = φ2deg(from.φ)
+
+        // create a to object
+        const to = {}
+        // clone object
+        Object.assign(to, this.activeMarker.location)
```
Comment (jurgenbelien) [inline components/globe-component/globe-component.js]: We have to remember to polyfill this one

File: components/globe-component/constants.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import * as THREE from 'three'
 
 export const GLOBE_RADIUS = 5
-export const INIT_ANGLE = Math.PI / 2
-export const INIT_AXIS = new THREE.Vector3(1, 0, 0)
+export const INIT_ANGLE = 0 // Math.PI / 2
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply components/globe-component/constants.js]: Replaced by setting camera.up.

File: components/globe-component/common.js [hunk 6]
```diff
@@ -1,19 +1,34 @@
...
-  const radius = Math.sqrt((x * x) + (y * y) + (z * z))
-  const latitude = Math.acos(z / radius)
-  const longitude = Math.atan2(y, x)
-
-  return { radius, latitude, longitude }
+  // radius r, inclination θ (latitude), azimuth φ (longitude)
+  const r = Math.sqrt((x * x) + (y * y) + (z * z))
+  const θ = Math.acos(z / r)
+  const φ = Math.atan2(y, x)
+  return { r, θ, φ }
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply components/globe-component/common.js]: replaced to {r, theta, phi}. Comment added with source of formulas (https://en.wikipedia.org/wiki/Spherical_coordinate_system)

File: components/globe-component/common.js [hunk 7]
```diff
@@ -25,3 +40,11 @@ export const lat2rad = scaleLinear()
 export const lon2rad = scaleLinear()
   .domain([180, -180])
   .range([0, 2 * Math.PI])
+
+export const θ2deg = scaleLinear()
+  .domain([0, Math.PI])
+  .range([-90, 90])
+
+export const φ2deg = scaleLinear()
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply components/globe-component/common.js]: replaced

File: components/globe-component/common.js [hunk 8]
```diff
@@ -25,3 +40,11 @@ export const lat2rad = scaleLinear()
 export const lon2rad = scaleLinear()
   .domain([180, -180])
   .range([0, 2 * Math.PI])
+
+export const θ2deg = scaleLinear()
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply components/globe-component/common.js]: replaced

File: components/globe-component/globe-component.js (hunk 4 above)
Comment (SiggyF) [reply components/globe-component/globe-component.js]: I would normally use _.clone from lodash. I see it's not supported by IE. Replaced by explicit assignment.

File: components/globe-component/globe-component.js (hunk 2 above)
Comment (SiggyF) [reply components/globe-component/globe-component.js]: replaced

File: components/globe-component/particles.js (hunk 3 above)
Comment (SiggyF) [reply components/globe-component/particles.js]: It's not gzipped at the moment. There's a few optimizations. It now contains:
- too many decimals
- data for all variables in 1 file
- uncompressed data

In this case I think we can encode the data in an image. I'll implement that in a separate issue/PR.

File: components/globe-component/common.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (SiggyF) [reply components/globe-component/common.js]: also renamed functions for clarity

File: components/globe-component/particles.js (hunk 3 above)
Comment (SiggyF) [reply components/globe-component/particles.js]: Put in the backlog as WWA-124

**Pull request merged**